pkg/tracegen/templated.go

```diff
@@ -82,6 +94,73 @@ type TraceTemplate struct {
 	Spans []SpanTemplate `js:"spans"`
 }
 
+type Link struct {
+	// LinkToPreviousSpanIndex true will set TraceID and SpanID the same as the previous span
+	LinkToPreviousSpanIndex bool `js:"linkToPreviousSpanIndex"`
+	// Attributes for this link
+	Attributes map[string]interface{} `js:"attributes"`
+	// Generate random attributes for this link
+	RandomAttributes *AttributeParams `js:"randomAttributes"`
+}
+
+type Event struct {
+	// Name of event
+	Name string `js:"name"`
+	// Attributes for this event
+	Attributes map[string]interface{} `js:"attributes"`
+	// Generate random attributes for this event
+	RandomAttributes *AttributeParams `js:"randomAttributes"`
+}
+
+type LinkDefaults struct {
+	// LinkToPreviousSpanIndex true will set TraceID and SpanID the same as the previous span
+	// Unless it is the first span then a random TraceID and a random SpanID will be used
+	LinkToPreviousSpanIndex bool `js:"linkToPreviousSpanIndex"`
+	// Rate of random links per each span
+	Rate float32 `js:"rate"`
+	// Generate random attributes for this link
+	RandomAttributes *AttributeParams `js:"randomAttributes"`
+}
+
+type EventDefaults struct {
```

Comment:
Same comment as for links about merging `EventDefaults` and `RandomEvents` to `EventParams`